pkg/labels/consul_aggregator.go

```diff
@@ -127,55 +131,21 @@ func (c *consulAggregator) Aggregate() {
 	for {
 		loopTime := time.After(c.aggregationRate)
 		select {
+		case err := <-outErrors:
+			c.logger.WithError(err).Errorln("Error during watch")
 		case <-c.aggregatorQuit:
 			return
 		case pairs := <-outPairs:
-			// Convert watch result to []Labeled
-			c.labeledCache = make([]Labeled, len(pairs))
-			for i, kvp := range pairs {
-				val, err := convertKVPToLabeled(kvp)
-				if err != nil {
-					c.logger.WithErrorAndFields(err, logrus.Fields{
-						"key":   kvp.Key,
-						"value": string(kvp.Value),
-					}).Errorln("Invalid key encountered, skipping this value")
-					continue
-				}
-				c.labeledCache[i] = val
-			}
+			c.watcherLock.Lock()
```

Comment:
actually this is already very concise i was just thrown off by the sea of red. don't worry about it